mesh/pmesh.cpp

```diff
@@ -5568,6 +5568,97 @@ void ParMesh::PrintSharedEntities(const char *fname_prefix) const
    }
 }
 
+void ParMesh::GetGlobalVertexIndices(Array<HYPRE_Int> &gi)
+{
+   H1_FECollection fec(1, Dim); // Order 1, mesh dimension (not spatial dimension).
+   ParFiniteElementSpace fespace(this, &fec);
+
+   MFEM_VERIFY(fespace.GetVSize() == GetNV(),
+               "There should be a bijection between DOFs and vertices");
```

Comment:
What's a situation that this kind of checks would fail? Is this to catch the case where the mesh is not finalized, i.e., could have extra vertices?

Reply:
Good question. I originally wrote this out of a habit of using many checks in my codes, instead of thinking about whether something is safe (it's easier to put in a check than to think carefully and be sure something will never fail). Maybe it is not necessary, but removing the check means we should be very confident the check would never fail. In other words, I agree it is likely an excessive check, but I also don't know the mesh class 100%, so I would worry about the case I never thought of. On the other hand, the check is not expensive. I am fine with removing it if no one sees a potential case where it may fail.

Reply:
`MFEM_ASSERT` seems more appropriate to me.